docs/config.rst

```diff
@@ -266,16 +266,15 @@ Configuration Options
 ``kafkastore.bootstrap.servers``
   A list of Kafka brokers to connect to. For example, `PLAINTEXT://hostname:9092,SSL://hostname2:9092`
 
-  If this configuration is not specified, the Schema Registry's internal Kafka clients will get their Kafka bootstrap server list
-  from ZooKeeper (configured with `kafkastore.connection.url`). Note that if `kafkastore.bootstrap.servers` is configured,
-  `kafkastore.connection.url` still needs to be configured, too.
+  If this configuration is not specified, the Schema Registry's internal Kafka clients will get their Kafka bootstrap server list from ZooKeeper (configured with `kafkastore.connection.url`). In that case, all available listeners matching the `kafkastore.security.protocol` setting will be used. Note that if `kafkastore.bootstrap.servers` is configured, `kafkastore.connection.url` still needs to be configured, too.
```

Comment:
I think that's less reliable. It's actually a good thing that brokers can be automatically loaded from ZK as it is fewer things the user has to configure and it also means fewer things they might need to update over time (e.g. as their set of bootstrap brokers change depending on how they manage hostnames and hardware). It's really only this special case where the ZK cluster used for coordination is different from the one used for Kafka.

Another consideration is whether we should also introduce another config. We're actually kind of abusing `kafkastore.connection.url` and reusing it (combined with `schema.registry.zk.namespace`) for ZookeeperMasterElector. This is obviously simpler and easier when you are just using the same ZK for both, but as we're seeing now, we've evolved to use cases where that isn't true. Another change that could be made is to introduce some new `schemaregistry.connection.url` and have it default to `kafkastore.connection.url`. In that case, users of a cloud Kafka would set `schemaregistry.connection.url`, *not* set `kafkastore.connection.url` and set `kafkastore.bootstrap.servers`. This is slightly nicer in that it might be a bit clearer which configs you have to set rather than having 2 sort of conflicting settings for bootstrap connections under the `kafkastore.` prefix.